Moved repeated constant declarations to a common location

kalyan-nidumolu opened the pull request:
Following up on moving all common constants to a common location instead of replicating them in each agent.

rvs commented:
What's the deal with this PR @nordmark and @kalyan-nidumolu ? I kind of would like to see it merged.

LGTM from my side, btw.

eriknordmark commented on pkg/pillar/cmd/diag/diag.go:
```diff
@@ -20,24 +28,11 @@ import (
 const (
-	agentName       = "diag"
-	identityDirname = "/config"
-	selfRegFile     = identityDirname + "/self-register-failed"
```
selfRegFile is only used by diag. So why promote it to common?

kalyan-nidumolu replied:
Hmm.. Who creates this file? grep doesn't seem to indicate ?

Hmm.. Should this really . be "self-register-pending" ?? That is the file being created by device-steps.sh. 

Hmm.. looking at this file - Client just doesn't create this file. It seems to be in this infinite loop trying to self register. Should diag be fixed??

eriknordmark replied:
Yes, it should be self-register-pending. Diag just looks at it for its output.

What infinite loop??

eriknordmark commented on pkg/pillar/cmd/diag/diag.go:
```diff
@@ -20,24 +28,11 @@ import (
-	serverFileName  = identityDirname + "/server"
-	deviceCertName  = identityDirname + "/device.cert.pem"
-	onboardCertName = identityDirname + "/onboard.cert.pem"
-	onboardKeyName  = identityDirname + "/onboard.key.pem"
```
Only used in diag? But you might be missing changed to client.go??

kalyan-nidumolu replied:
Good Catch.. Hmm.. Client.go has these defined relative to IdentityDir  And default value of IdentityDir is /config. But client has an option to pass on identityDir ( -D option ) that the user can theoretically set to something other than /config. But diag has this hardcoded to /config.

I am assuming we no longer need this flexibility to use different identityDirs.. and finalize on /config?

eriknordmark replied:
Dropping the command line option to client is fine. I assume any testing of full agents will be done in a chroot/container when we get to that.

kalyan-nidumolu replied:
Done.. Removed -D option from Client.go

eriknordmark commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -4,21 +4,62 @@
 	TmpDirname = "/var/tmp/zededa"
-	// PersistDir - Persist Directory
+	// AADirname - directory for per-model Assignable Adapters files
+	AADirname = TmpDirname + "/AssignableAdapters"
```
This doesn't exist any more (still exists in build but isn't used)

kalyan-nidumolu replied:
Removed it..

eriknordmark commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -4,21 +4,62 @@
+	// AADirname - directory for per-model Assignable Adapters files
+	AADirname = TmpDirname + "/AssignableAdapters"
+	// DNCDirname - directory for per-model Network Config files
+	DNCDirname = TmpDirname + "/DeviceNetworkConfig"
```
Ditto

kalyan-nidumolu replied:
Removed it..

eriknordmark commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -4,21 +4,62 @@
-	// RwImgDirname - location for images
-	RwImgDirname = PersistDir + "/img" // We store images here
-	// DownloadDirname - Location for downloader to download images
+	// RwImgDirname - We store images here
```
Only used  by domainmgr, hence not common.

kalyan-nidumolu replied:
moved it back to domainmgr

eriknordmark commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -4,21 +4,62 @@
+	// CertObj - Name of Certificate obj. dir
+	CertObj = "cert.obj"
+	// ConfigDir - name of Config dir.
+	ConfigDir = "/config"
```
But you call this IdentityDirname as well? Pick one?

kalyan-nidumolu replied:
I followed the existing code.. I thought it was intentional - to have 2 variables - identityDir and ConfigDir..  if not, I would pick "ConfigDir" as the name ( and not IdentityDir )..

kalyan-nidumolu commented:
> Looking at a bunch of these but not all, turns out that many of them are only used by one agent. Hence it doesn't make sense to promote them to a common place.
> If you look at client.go you'll find some more shared with diag.go but they are not used in more than those.
> And at least two have become unused since you started this.
> 
> The appimg/baseosimg/certobj enum's (expressed as strings) can definitely be put in one place.
> And picking one name for /config (as opposed to two different ones; IdentityDirname makes more sense than ConfigDirname IMHO) makes sense.

Moved back the following to individual files since they are local:

1) RwImgDirname = PersistDir + "/img"
2) checkpointDirname
3)  PendingDirname = ImgCatalogDirname + "/pending"
4) FinalDirname = ImgCatalogDirname + "/verified"
5) PersistRktLocalConfigDir = PersistDir + "/rktlocal"
6) PersistRktLocalConfigAuthDir = PersistRktLocalConfigDir + "/auth.d"

Moved some others to the agent if they are only being used by 1 agent.

Similar to /config, the following location also has 2 names:

	FinalDirname = ImgCatalogDirname + "/verified"
	// VerifiedDirname - Location of verified App images. Read-only images
	// named based on sha256 hash each in its own directory
	VerifiedDirname = ImgCatalogDirname + "/verified"

which one is better? "VerifiedDirName" seems to be the better name??

eriknordmark commented on pkg/pillar/cmd/verifier/verifier.go:
```diff
@@ -47,12 +47,8 @@ const (
-	rootCertDirname    = "/config"
-	rootCertFileName   = rootCertDirname + "/root-certificate.pem"
-	certificateDirname = persistDir + "/certs"
+	rootCertDirname  = "/config"
```
I would think this should be the global identityDirname you put in the shared file and nothing else.

kalyan-nidumolu replied:
Done.. Removed it..

eriknordmark commented on pkg/pillar/cmd/zedagent/handlelookupparam.go:
```diff
@@ -30,8 +30,8 @@ import (
 const (
-	infraFileName           = identityDirname + "/infra"
-	zedserverConfigFileName = tmpDirname + "/zedserverconfig"
+	infraFileName           = types.IdentityDirname + "/infra"
```
Is this used? There is no /config/infra on running devices.

kalyan-nidumolu replied:
It is being used in Zedagent... looks like provides a backdoor:

	ACLPromisc := false
	if _, err := os.Stat(infraFileName); err == nil {
		log.Debugf("Setting ACLPromisc\n")
		ACLPromisc = true
	}

eriknordmark commented on pkg/pillar/cmd/zedmanager/zedmanager.go:
```diff
@@ -24,11 +24,8 @@ import (
-
-	certificateDirname = persistDir + "/certs"
+	agentName    = "zedmanager"
+	finalDirname = types.ImgCatalogDirname + "/verified"
```
You've already define VerifiedDirname so please use it.

kalyan-nidumolu replied:
Done..

eriknordmark commented on pkg/pillar/types/locationconsts.go:
```diff
@@ -4,21 +4,44 @@
-	// VerifiedDirname - Read-only images named based on sha256 hash each
-	// in its own directory
+	// ObjectDownloadDirname - Location of downloaded objects
+	ObjectDownloadDirname = PersistDir + "/downloads"
```
On line 13 you have a different name for the same directory. There should be only one name for each directory.

kalyan-nidumolu replied:
Hmm.. Thinking was.. The same directory could be used for different purposes.. So could use different names.. Fixed it by doing:
   ObjectDownloadDirname = DownloadDirname

Will look through some of these usages to see if it good to replace ObjectDownloadDirname with DownloadDirname...

eriknordmark commented:
@kalyan-nidumolu are you planning to take care of the duplicate names for the same constant so we can pull this?

kalyan-nidumolu commented:
Done..  Eliminated all direct duplicates.. There are a few derived ones that can be replaced as well, but will do that in a separate commit. Lets get this one in - The diff is getting bigger..

eriknordmark commented:
On 10/16/19 12:22 PM, kalyan-nidumolu wrote:
> *@kalyan-nidumolu* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In pkg/pillar/types/locationconsts.go 
> <https://github.com/lf-edge/eve/pull/254#discussion_r335664117>:
> 
>>  	DownloadDirname = PersistDir + "/downloads"
> -	// ImgCatalogDirname - Location for App Images
> -	ImgCatalogDirname = DownloadDirname + "/" + appImgObj
> -	// VerifiedDirname - Read-only images named based on sha256 hash each
> -	// in its own directory
> +	// ObjectDownloadDirname - Location of downloaded objects
> +	ObjectDownloadDirname = PersistDir + "/downloads"
> 
> Hmm.. Thinking was.. The same directory could be used for different 
> purposes.. So could use different names.. Fixed it by doing:
> ObjectDownloadDirname = DownloadDirname

That argument doesn't make any sense to me. Everything we download is an 
object. No reason to have two different global names for the same 
directory.

> 
> Will look through some of these usages to see if it good to replace 
> ObjectDownloadDirname with DownloadDirname...

Please replace them all so we can get this PR done.

    Erik

> 
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub 
> <https://github.com/lf-edge/eve/pull/254?email_source=notifications&email_token=AGNLQWAJBX2MPM2P3XOZHG3QO5SXFA5CNFSM4IUQRW22YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCIGKH7I#discussion_r335664117>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AGNLQWE2NIOQESA7K7OTAC3QO5SXFANCNFSM4IUQRW2Q>.
>

kalyan-nidumolu commented:
> On 10/16/19 12:22 PM, kalyan-nidumolu wrote: ***@***.**** commented on this pull request. ------------------------------------------------------------------------ In pkg/pillar/types/locationconsts.go <[#254 (comment)](https://github.com/lf-edge/eve/pull/254#discussion_r335664117)>: > DownloadDirname = PersistDir + "/downloads" - // ImgCatalogDirname - Location for App Images - ImgCatalogDirname = DownloadDirname + "/" + appImgObj - // VerifiedDirname - Read-only images named based on sha256 hash each - // in its own directory + // ObjectDownloadDirname - Location of downloaded objects + ObjectDownloadDirname = PersistDir + "/downloads" Hmm.. Thinking was.. The same directory could be used for different purposes.. So could use different names.. Fixed it by doing: ObjectDownloadDirname = DownloadDirname
> That argument doesn't make any sense to me. Everything we download is an object. No reason to have two different global names for the same directory.
> Will look through some of these usages to see if it good to replace ObjectDownloadDirname with DownloadDirname...
> Please replace them all so we can get this PR done. Erik
> […](#)
> — You are receiving this because your review was requested. Reply to this email directly, view it on GitHub <#254?email_source=notifications&email_token=AGNLQWAJBX2MPM2P3XOZHG3QO5SXFA5CNFSM4IUQRW22YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCIGKH7I#discussion_r335664117>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AGNLQWE2NIOQESA7K7OTAC3QO5SXFANCNFSM4IUQRW2Q>.

Done..
